code/modules/power/treadmill.dm

```diff
@@ -40,7 +40,7 @@
 
 /obj/machinery/power/treadmill/proc/throw_off(atom/movable/A)
 	// if 2fast, throw the person, otherwise they just slide off, if there's reasonable speed at all
-	if(speed)
+	if(speed && !A.move_resist == INFINITY)
```

Comment:
I'm not sure why this isn't being caught by anything in the CI, but it looks like this needs to have some extra brackets around it.
Currently it doesn't actually throw *anything* off of the treadmill, so I assume it's reading the line as this:
```dm
if((speed && !A.move_resist) == INFINITY)
```
Rather than this:
```dm
if(speed && !(A.move_resist == INFINITY))
```
Specifying that it's the second one fixes the issue:
```suggestion
	if(speed && !(A.move_resist == INFINITY))
```